File: RedSheet.DbEntities/Models/Personality.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("Personalities", Schema = "dbo")]
	[Serializable]
	public partial class Personality  
	{
 
	[MaxLength(1000)]
	public string Description { get; set; } 
 
	[Required] 
	[MaxLength(50)]
	public string PersonalityColor { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int PersonalityId { get; set; } 
 
	[MaxLength(50)]
	public string PersonalityKey { get; set; } 
 
	[MaxLength(10)]
	public string PersonalityKeyText { get; set; } 
	[InverseProperty("Personality")]
	public ICollection<TheirTeamMember> TheirTeamMembers { get; set; } 


	public Personality()
	{
	this.TheirTeamMembers = new HashSet<TheirTeamMember>();

    }

    }
}

File: redsheet-portal/src/app/database-models/personality.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { TheirTeamMember,  } from './'
export class Personality {
    constructor(personality?: Personality )  {
        let properties = [ "description", "personalityColor", "personalityId", "personalityKey", "personalityKeyText", "theirTeamMembers",];
        for (let property of properties)
            if (personality && personality[property])
                this[property] = personality[property];
    }
 
    @maxLength(1000)
	description : string =   undefined;
 
    @required()
    @maxLength(50)
	personalityColor : string =   undefined;
 
	personalityId : number =   0 ;
 
    @maxLength(50)
	personalityKey : string =   undefined;
 
    @maxLength(10)
	personalityKeyText : string =   undefined;
	@nested(TheirTeamMember)
	theirTeamMembers: TheirTeamMember[];



}
